### PR TITLE
track staking events history; add staking rewards to validator history

### PR DESCRIPTION
**Pull request by Andrew7234**
This PR includes a chunk of the changes necessary for staking rewards that would be nice to include in the upcoming reindex release. There are two changes:
- runtime analyzer now tracks staking event history in fast-sync mode
- validator history analyzer now tracks per-validator staking rewards.

### CODE REVIEW

File: analyzer/validatorstakinghistory/validatorstakinghistory.go [hunk 1]
```diff
@@ -148,6 +148,17 @@ func (p *processor) ProcessItem(ctx context.Context, batch *storage.QueryBatch,
 			validators[vID] = struct{}{}
 		}
 	}
+	// Download staking events for the first block, which includes validator staking rewards.
+	stakingEvents, err := p.source.StakingEvents(ctx, epoch.startHeight)
+	if err != nil {
+		return fmt.Errorf("downloading staking events for height %d", epoch.startHeight)
```
Comment (pro-wh) [inline analyzer/validatorstakinghistory/validatorstakinghistory.go:154]: woa a lot of places in this file discard err

File: analyzer/validatorstakinghistory/validatorstakinghistory.go [hunk 2]
```diff
@@ -171,6 +182,14 @@ func (p *processor) ProcessItem(ctx context.Context, batch *storage.QueryBatch,
 			acct.Escrow.Debonding.TotalShares,
 			len(delegations),
 		)
+		// Update staking rewards for this validator in the previous epoch.
+		if ev, exists := stakingRewards[addr]; exists {
+			batch.Queue(queries.ValidatorStakingRewardUpdate,
+				vID.String(),
+				epoch.epoch-1, // previous epoch
```
Comment (pro-wh) [inline analyzer/validatorstakinghistory/validatorstakinghistory.go:189]: can you affirm that this is all right

1. staking rewards appear in the first block of the next epoch

File: analyzer/validatorstakinghistory/validatorstakinghistory.go (hunk 1 above)
Comment (Andrew7234) [reply analyzer/validatorstakinghistory/validatorstakinghistory.go:154]: good eye, fixed throughout

File: analyzer/validatorstakinghistory/validatorstakinghistory.go (hunk 2 above)
Comment (Andrew7234) [reply analyzer/validatorstakinghistory/validatorstakinghistory.go:189]: Hm I asked a follow up in core-platform but as of now, the e2e integration tests show that the staking rewards are indeed populated in the first block of the next epoch. However, I'm not sure if there are any failure edge cases where the rewards might be populated in a subsequent block; will ask around.

File: analyzer/validatorstakinghistory/validatorstakinghistory.go [hunk 3]
```diff
@@ -148,6 +148,30 @@ func (p *processor) ProcessItem(ctx context.Context, batch *storage.QueryBatch,
...
+	if err != nil {
+		return fmt.Errorf("downloading staking events for height %d: %w", epoch.startHeight, err)
+	}
+	stakingRewards := make(map[nodeapi.Address]nodeapi.AddEscrowEvent)
+	for _, e := range stakingEvents {
+		if e.StakingAddEscrow != nil && e.StakingAddEscrow.NewShares != nil && e.StakingAddEscrow.NewShares.IsZero() {
+			stakingRewards[e.StakingAddEscrow.Escrow] = *e.StakingAddEscrow
+		}
+	}
+	// EXPERIMENT: Download staking events for the second block, which includes validator staking rewards.
```
Comment (pro-wh) [inline analyzer/validatorstakinghistory/validatorstakinghistory.go:162]: so we'll go without this?

**Review by pro-wh — APPROVED** [review]
thanks for the changes
